### ISSUE TITLE
Can't load any YouTube videos

### ISSUE THREAD

**Issue by MooBloo22** [futo-org/grayjay-android#1231]
### What happened?

The video buffers for a while and a message saying 'detecting YouTube blocking attempt, fallback to iOS(1)' pops up before I get the error message telling me to login to prove I'm not a bot despite being logged in.
![Screenshot_2024-08-07-14-28-47-41_ca2ad563b57b5497046450f16b7718e2](https://github.com/user-attachments/assets/3449beaa-c42e-4d21-814a-ba6d5ccf4e0f)

### Grayjay Version

253

### What plugins are you seeing the problem on?

Youtube

### Plugin Version

190

### When do you experience the issue?

- [X] While logged in
- [X] While logged out
- [ ] N/A

**Comment by nOmnomNOM3** [comment]
Having same issue. Tried different accounts and clearing caches. Loads other sources fine

**Comment by codebymatter** [comment]
Definitely seeing the "Blocking attempt..Fallback to iOS" but have yet to encounter the message you've shared as img. Need to also mention that I watched a livestream for 45mins and did not encounter any errors. Noticed the msgs yesterday by the way.

**Comment by MooBloo22** [comment]
In my instance it had been occuring for hours before I posted and it seems to have 'scared' it into working again but it does appear I'm not the only one to have experienced this and it's an ongoing issue. Didn't do anything aside from clearing data, reinstalling, restarted my phone etc. Unsure what could have changed in the meantime.

**Comment by MooBloo22** [comment]
> In my instance it had been occuring for hours before I posted and it seems to have 'scared' it into working again but it does appear I'm not the only one to have experienced this and it's an ongoing issue. Didn't do anything aside from clearing data, reinstalling, restarted my phone etc. Unsure what could have changed in the meantim

**Comment by MooBloo22** [comment]
Found out the issue is very likely to be a conflict with my work's WiFi. I was wracking my mind for variables and it worked at home and on mobile data but not with my work's WiFi.

**Comment by nOmnomNOM3** [comment]
Having issues on cell and home wifi. YouTube work independently I can load
channels and content but I can't download or watch anything on YouTube. In
the past I had this issue when I did not sign in with the plugin. But that
is not the case anymore

**Comment by codebymatter** [comment]
> Found out the issue is very likely to be a conflict with my work's WiFi. I was wracking my mind for variables and it worked at home and on mobile data but not with my work's WiFi.

Welp that could be it I'm including acreens of me logging off and logging back on tho:

![Screenshot_20240808-085846_Grayjay](https://github.com/user-attachments/assets/a34ab9d1-356b-4446-bf2a-aed11a57f304)
![Screenshot_20240808-085858_Grayjay](https://github.com/user-attachments/assets/bb035465-a724-4ffd-8264-92a3f1f50ec5)
![Screenshot_20240808-085905_Grayjay](https://github.com/user-attachments/assets/eeeab7ca-3bd3-4de7-a03e-466205b561a0)
![Screenshot_20240808-085920_Grayjay](https://github.com/user-attachments/assets/9ffa4892-5523-4a88-9157-517a75878690)
![Screenshot_20240808-085948_Grayjay](https://github.com/user-attachments/assets/2a0a46bc-7b40-4279-b356-ac03d19b3f7d)
![Screenshot_20240808-090356_Grayjay](https://github.com/user-attachments/assets/8dc4788e-4558-4bd3-8529-70c1b93fb7a5)
![Screenshot_20240808-090521_Grayjay](https://github.com/user-attachments/assets/396506dd-6fd1-4d7c-b47d-f95c1e318992)

**Comment by nOmnomNOM3** [comment]
not a gj issue apparently. Turned off my VPN and videos load. I really hate
Google.

**Comment by Argel-Tal** [comment]
Happening for me too, no VPNs active. 

Triggers on each vid, sometimes it gets through the first warning msg, but will restart the vid. Other times it'll just keep throwing the warning after a short 5s buffering window. 

Only had had the login screen pop-up once

**Comment by MooBloo22** [comment]
![Screenshot_2024-08-09-14-13-38-61_ca2ad563b57b5497046450f16b7718e2](https://github.com/user-attachments/assets/8ab92619-b512-4405-a61b-8b97e95c8ba9)
New error message today. Still working expect when on work WiFi.

**Comment by MooBloo22** [comment]
![Screenshot_2024-08-09-14-34-32-97_ca2ad563b57b5497046450f16b7718e2](https://github.com/user-attachments/assets/5968e3dd-6ea5-46ea-9959-febd6c03238e)
![IMG_20240809_143749](https://github.com/user-attachments/assets/c9caa4b2-6269-4ba4-a8ce-78788b347e57)

**Comment by codebymatter** [comment]
Yes this looks a lot like something I woyld see when on a VPN. The Google Bot Check can also happen when Google dstects a different IP address or network deemed as suspicious. Not sure how to technically phrase it, sorry.. This sometimes happens on Private Browsing also.

My previous employer had free public WiFi and I remember connecting once to it so I can stream at a higher resolution and when I returned home, Google detectef a suspicious connection, requiring captcha, even thougb my home neteork is always used.

**Comment by Argel-Tal** [comment]
![Screenshot_20240810_123430_Grayjay.jpg](https://github.com/user-attachments/assets/19189f07-320d-48ff-9416-0a688f1d7f51)

Happened again today, even with "login for video/channel content" and "provide youtube activity" swtiched off


Version information (version_name = 253, version_code = 253, flavor = stable, build_type = release)
Device information (brand= samsung, manufacturer = samsung, device = a32x, version-sdk = 33, version-os = samsung/a32xxx/a32x:13/TP1A.220624.014/A326BXXU9CWL1:user/release-keys)

(e, ExceptionActivity, 2024-08-10 02:34:18): Uncaught exception ("Unknown Context"): [Youtube] Login required
Reason: Sign in to confirm you’re not a bot (455)[2-3]

**Comment by prkuctd** [comment]
Is it related to sponsorblock? When I disable SponsorBlock, I do not need to prove myself not to be a bot.

**Comment by MuriFlavo** [comment]
> Is it related to sponsorblock? When I disable SponsorBlock, I do not need to prove myself not to be a bot.

I have never used sponsorblock and still get it with vpn on, vpn off it still falls back to iOS.

**Comment by serossi** [comment]
its very likely not VPN related directly. its https, so it shouldnt make any difference.

however youtube always rollsout changes in batches, based on region and IP blocks. so what we see here is either a testrun or a first batch of an release.
given the recent crackdown on ublock on chrome i would guess its a rollout of a new anti adblock / anti 3rd party client wave

**Comment by MooBloo22** [comment]
![Screenshot_2024-08-22-20-10-18-14_ca2ad563b57b5497046450f16b7718e2](https://github.com/user-attachments/assets/c5d607d7-4384-4004-9e17-074df8268edf)
Latest version of this error after the most recent update.

**Comment by TheShanMan** [comment]
Grayjay has become unusable for me on a vpn because YT thinks I'm a bot. I wish it had the ability to use Invidious instances. That's what's been working for me.

I disagree that it's not specifically vpn related. Some sites block vpn's entirely, and others make you jump through captcha hoops. The internet is becoming unfriendly to anonymity. I hope Grayjay will embrace privacy minded folks and implement fixes/workarounds for use cases like vpn's.

**Comment by dimyself** [comment]
same for me today after i just updated the youtube plugihn

every video i click on in youtube is like this today.

i just updated the youtube plugin (version 194)

"Use login for channels" is off

"allow loginfallback" is on

"use login for video details" is off

"provide youtube activity" is off

"allow age / controversial restricted" are both on

"sponsorblock" is enabled

every single youtube video i click on now says:

"sign in to confirm you're not a bot (415). failed to load video"

**Comment by codebymatter** [comment]
Good question. I wouldn't be worried if I dowbloaded YT videos through Grayjay honestly like in the same way as using YT to MP3, which I rarely do much of.

I am a VPN user and understand what could be described or discerned as an extreme of leaving little to no user footprint fine but I must ask: Is this a relative issue with ReVanced? Newpipe? Do these work flawlessly with an active VPN or LAN proxy?

**Comment by TheShanMan** [comment]
For me, Newpipe and Revanced have the same issue, as does youtube itself opened in a browser. The one thing that still does work: Invidious, and that's because they've come up with a strategy to overcome this issue. I don't know if the strategy they're using could be applied to apps because they apply the strategy on the server side, but if nothing else it would be great if Grayjay had a mechanism for specifying an invidious instance (ideally, multiple instances) and using that for youtube requests.

I haven't been following current activity on Newpipe or Revanced, so I don't know if they're coming up with solutions to this problem.

**Comment by support-grayjay** [comment]
@MooBloo22 Could you try and retest it with the latest version of Grayjay (v264) and Youtube (209)?

**Comment by MooBloo22** [comment]
> @MooBloo22 Could you try and retest it with the latest version of Grayjay (v264) and Youtube (209)?

It works. Kind of. The first video will load but the second one says 'failed to load ump, restart plugin/app.'

**Comment by TheYoungerSes** [comment]
> @MooBloo22 Could you try and retest it with the latest version of Grayjay (v264) and Youtube (209)?

User from Canada, currently on Grayjay 264, and has Youtube on 209. I have Grayjay installed on two devices, a Samsung Galaxy S10, and a Moto G Power from 2020. On my home network, everything works fine. I can access content via IOS or UMP. But at work, which uses a VPN for the entire warehouse, I cannot access content via UMP, and IOS does not app eee as as an option. However, I can access content via reverting to version 204. Hopefully this can give some light as to how best fix this problem.

**Comment by Zvonimir-FUTO** [comment]
@TheYoungerSes This seems like issue is related to VPN which we already encountered few of those and will have to be looked at.

**Comment by Zvonimir-FUTO** [comment]
Hi everyone, UMP has been finally fixed and new version of Grayjay and Youtube have been released!

Go update and test and tell us if you find issues except the one below.

We are aware that age restricted, controversial videos, or other videos that require login may not yet work. We will be working on this next week.

The desktop release that is required for this fix will also be released next week.

**Comment by dimyself** [comment]
Installing now... what is UMP? Thanks!

**Comment by dimyself** [comment]
> Hi everyone, UMP has been finally fixed and new version of Grayjay has been released!
> 
> Go update and test and tell us if you find issues except the one below.
> 
> We are aware that age restricted, controversial videos, or other videos that require login may not yet work. We will be working on this next week.
> 
> The desktop release that is required for this fix will also be released next week.

I just tried the new version... i'm still getting 'login required' for most ip's using Nordvpn

Maybe thats not what this fix was for?

Are there any ways to fix vpn ip blocking?

Is there any vpn list for youtube that shows whatS working and whats not? 

I can say most of the Nordvpn ip's in almost all countries are blocked by youtube (at least for me)! (login required error)

I'm not logging into youtube account

**Comment by Zvonimir-FUTO** [comment]
@dimyself Original issue was not VPN related. The UMP format is used by YouTube for a number of requests and responses.

**Comment by TheShanMan** [comment]
Yeah I'd love to see a solution for vpn users because grayjay is unfortunately useless to me currently because of this problem. FWIW I'm currently using a dedicated web browser to load youtube.com with a vpn location they're not yet blocking for their site, but this vpn location doesn't work with grayjay (or newpipe).

**Comment by Zvonimir-FUTO** [comment]
There is issue #1334 already opened for it. We will look into it eventually but these are not easy to reproduce and troubleshoot.

**Comment by dimyself** [comment]
> Yeah I'd love to see a solution for vpn users because grayjay is unfortunately useless to me currently because of this problem. FWIW I'm currently using a dedicated web browser to load youtube.com with a vpn location they're not yet blocking for their site, but this vpn location doesn't work with grayjay (or newpipe).

youtube site (in private mode or logged in) works fine using vpn on my desktop. 

But  youtube apps (grayjay, newpipe, etc) require login on phone and desktop.

**Comment by nOmnomNOM3** [comment]
It seems to ve worse now. It was working fine for me and now seemingly
every video stops at 59 again. It did ask me to sign in the first video I
tried to watch earlier after the patch was pushed. And since then ive had
trouble. This is with gj excepted from my vpn (gj traffic not tunneled)

**Comment by nOmnomNOM3** [comment]
If I restart the video and skip past 0:59 then the rest of the video plays

**Comment by Zvonimir-FUTO** [comment]
@nOmnomNOM3 please follow and comment on thread https://github.com/futo-org/grayjay-android/issues/1807. There are already too many issues for this opened and hard to track everything.
